cmake/SanitizerOptions.cmake

```diff
@@ -0,0 +1,36 @@
+if (ENABLE_SANITIZERS)
+  set(ENABLE_ADDRESS_SANITIZER ON)
+  set(ENABLE_UNDEFINED_SANITIZER ON)
+endif()
+
+# Inlclude build macros for updating configuration variables
+include(HandleLibcxxFlags)
+
+set (SANITIZER_FLAGS_LIST "")
+set (SANITIZER_EXE_LINKER_FLAGS_LIST "")
+set (SANITIZER_SHARED_LINKER_FLAGS_LIST "")
+
+if(ENABLE_ADDRESS_SANITIZER)
+  message(STATUS "Enabling address sanitizer (ASan).")
+  set(OLD_CMAKE_REQUIRED_FLAGS ${CMAKE_REQUIRED_FLAGS})
+  set(CMAKE_REQUIRED_FLAGS "-fsanitize=address")
```

Comment:
You can also "append" to the flags variable with:
```
set(CMAKE_REQUIRED_FLAGS "${CMAKE_REQUIRED_FLAGS} -fsanitize=address")
```
I haven't tried it, but I _think_ it should work.

Reply:
It does work. It also looks safer to me so I'll use it, thanks